OpenRA.Mods.Common/Widgets/SupportPowerTimerWidget.cs

```diff
@@ -23,25 +23,34 @@ public class SupportPowerTimerWidget : Widget
 		public readonly string Format = "{0}: {1}";
 		public readonly TimerOrder Order = TimerOrder.Descending;
 
-		readonly World world;
+		readonly int timestep;
 		readonly IEnumerable<SupportPowerInstance> powers;
 		Pair<string, Color>[] texts;
 
 		[ObjectCreator.UseCtor]
 		public SupportPowerTimerWidget(World world)
 		{
-			this.world = world;
 			powers = world.ActorsWithTrait<SupportPowerManager>()
 				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant)
 				.SelectMany(s => s.Trait.Powers.Values)
 				.Where(p => p.Instances.Any() && p.Info.DisplayTimer && !p.Disabled);
+
+			// Timers in replays should be synced to the effective game time, not the playback time.
+			timestep = world.Timestep;
+			if (world.IsReplay)
+			{
+				GameSpeed speed;
+				var gameSpeeds = Game.ModData.Manifest.Get<GameSpeeds>();
+				if (gameSpeeds.Speeds.TryGetValue(world.LobbyInfo.GlobalSettings.GameSpeedType, out speed))
+					timestep = speed.Timestep;
+			}
```

Comment:
These cases are subtly (and importantly) different, so there is no meaningful amount of code that can be reused.